app/controllers/widgets_controller.rb

```diff
@@ -0,0 +1,56 @@
+# app/controllers/widget_controller.rb:
+# Handle widgets, if enabled
+#
+# Copyright (c) 2014 UK Citizens Online Democracy. All rights reserved.
+# Email: hello@mysociety.org; WWW: http://www.mysociety.org/
+
+class WidgetsController < ApplicationController
+
+    require 'securerandom'
+
+    before_filter :find_info_request, :check_widget_config, :check_prominence
```

Comment:
Check that the widget is enabled first so that we avoid db queries if the feature is not enabled?